### ISSUE TITLE
[TF2] Cannot play workshop maps

### ISSUE THREAD

**Issue by piluke** [ValveSoftware/Source-1-Games#2066]
I have TF2 on a different disk than my Steam installation and SteamApps folder. When I join a server which is running a workshop map, it successfully downloads the map and then gives me a popup with "Missing map /path/to/map.bsp,  disconnecting" If I check that path, the file exists and I can copy it to tf/maps and run it fine.

[Here is the full log from my console.](https://gist.github.com/piluke/424d963abff5a11b38d51de98346b93a)

[Here is my system information.](https://gist.github.com/piluke/8c04b9c24ab151fda406968a121bf2d6)

**Comment by Haze33E** [comment]
All the maps have to be in tf/maps in order to play them. TF2 doesn't have workshop support for downloading and playing maps immediately like garry's mod does. You have to either move or copy the map into the tf/maps directory every time you download a new map. But if you join a server running a map you don't have it downloads it to the tf/maps directory for you. So basically it's just a lack of workshop integration on Valve's part.

**Comment by hahagoni** [comment]
I am using a Windows system, this is most likely a multiplatform issue. Haze's info is right. Proper workshop integration is needed.

**Comment by Haze33E** [comment]
Maybe it's just a Linux thing then. I'm on Linux too and I think I tried the command once and it didn't work so I've just moved the maps ever since. At least that's how I remember it.

**Comment by piluke** [comment]
The main issue for me is that if the server is running a workshop map, even if I move the downloaded map to tf/maps, the server will not let me join because it can't find the map that it downloaded. That is why I believe it might be a filesystem issue.

**Comment by Haze33E** [comment]
Was this after a recent update? You might have corrupted files or something.

**Comment by piluke** [comment]
No, this has been an issue for at least a couple months, maybe longer. I just tested it with another computer running Ubuntu 16.04 and it worked. On that computer, the entire SteamApps folder is on a different filesystem than the rest of the Steam installation.

This makes sense because TF2 downloads the maps to SteamApps/workshop/content/440/ and in this second case, they are on the same filesystem.

**Comment by piluke** [comment]
On my original computer, I moved only the SteamApps/workshop folder to the same filesystem as the tf/ folder and it works correctly.

**Comment by agrastiOs** [comment]
Interesting. That means workshop folder is only used for TF2 workshop maps

**Comment by agrastiOs** [comment]
So I guess the problem is that TF2 doesn't find the workshop folder (because it's in a different drive), and thus doesn't load the map. It should be made so TF2 could see the folder in the different drive or just download the workshop maps to a different location, like tf/maps/workshop

**Comment by agrastiOs** [comment]
@kisak-valve this is a problem for all platforms

**Comment by Nephyrin** [comment]
When you encounter this, is the "/path/to/cp_glassworks_rc6a.bsp" that is printed definitely an existing file, including case sensitivity?  Are either of the involved filesystems network filesystems or similar?
